### PR TITLE
Support Windows

### PR DESCRIPTION
**Pull request by marcelm**
- The GitHub Actions job that created the wheels needed to become a matrix job that runs on Windows and Linux. Because the sdist was also created in that job and I did not want it to be created multiple times, sdist creation, wheel creation and publishing are now separate jobs.
- `pipx` is installed on the Linux runners, so building the sdist could be simplified to `pipx run build` (and to test whether a wheel can be created from the sdist, I do not add the `--sdist` option).
- The other changes were just the things necessary to make the code work on both platforms.
- The Windows tests should no longer fail as soon as https://github.com/pycompression/xopen/pull/83 is merged and a new xopen version is published.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (marcelm) [inline .github/workflows/ci.yml:63]: Before merging, this needs to be uncommented again. I just want to see the generated wheels once.

**Comment by rhpvorderman** [comment]
> The Windows tests should no longer fail as soon as Add seek() and tell() to PipedCompressionReader pycompression/xopen#83 is merged and a new xopen version is published.

I tested a solution with peek only, but indeed that does not work on io.BytesIO. It is a bit weird that a homebrew version of peek has to be implemented, but alas. I have reviewed the PR on xopen and it can be merged!

**Comment by rhpvorderman** [comment]
I recommend disabling the rest of the testing (temporarily) so you get the wheelbuilding results faster.

**Comment by marcelm** [comment]
> I recommend disabling the rest of the testing (temporarily) so you get the wheelbuilding results faster.

That’s easy to say afterwards :smile:! At each commit, I was hoping that it would be the last one ...

**Comment by marcelm** [comment]
Stupid drive letters.

**Comment by marcelm** [comment]
Ah, I like it how a rebase makes it appear as I had exactly known what to do from the beginning ...

File: .github/workflows/ci.yml
Comment (rhpvorderman) [reply .github/workflows/ci.yml:63]: @marcelm. Don't forget!

**Review by rhpvorderman — APPROVED** [review]

**Comment by rhpvorderman** [comment]
The code looks good to me. Except the part that was commented out and shouldn't be forgotten :wink:. 
I am glad we can support windows now! Thank you for doing all the very very hard work of CIBuildWheel testing.

**Comment by marcelm** [comment]
> Whoops, still in xopen-mode where I am considered co-maintainer. How is that here? I don't mean to be pretentious by approving you on your own project.

That’s fine (and I’m glad for the feedback), feel free to act like a co-maintainer! (At the moment, I still feel ok with committing minor things without going through a PR, though, but that may change.)

**Comment by marcelm** [comment]
Sorry I bumped the cibuildwheel version, another 15 minute delay.

**Comment by rhpvorderman** [comment]
> That’s fine (and I’m glad for the feedback), feel free to act like a co-maintainer! (At the moment, I still feel ok with committing minor things without going through a PR, though, but that may change.)

Thanks for the clarification. It is your project, so my interference will indeed mostly be feedback on the more substantial things. I am very much planning to do everything trough PR myself.

**Pull request merged**
